DSCResources/MSFT_xNetBIOS/MSFT_xNetBIOS.psm1

```diff
@@ -0,0 +1,137 @@
+    Add-Type -TypeDefinition @'
+    public enum NetBiosSetting
+    {
+       Default,
+       Enable,
+       Disable
+    }
+'@
+}
+
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    try
+    {
+        $Netadapterparams = @{
+            ClassName = 'Win32_NetworkAdapter'
+            Filter = 'NetConnectionID="{0}"' -f $InterfaceAlias
+        }
+        $NetAdapterConfig = Get-CimInstance @Netadapterparams | 
+            Get-CimAssociatedInstance -ResultClassName Win32_NetworkAdapterConfiguration
+    }
+    catch
+    {
+        throw $_
+    }
+    return @{
+        InterfaceAlias = $InterfaceAlias
+        Setting = $([NETBIOSSetting].GetEnumValues()[$NetAdapterConfig.TcpipNetbiosOptions])
+    }
+}
+
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $Netadapterparams = @{
+        ClassName = 'Win32_NetworkAdapter'
+        Filter = 'NetConnectionID="{0}"' -f $InterfaceAlias
+    }
+    $NetAdapterConfig = Get-CimInstance @Netadapterparams | 
+        Get-CimAssociatedInstance -ResultClassName Win32_NetworkAdapterConfiguration
+
+    if ($Setting -eq [NETBIOSSetting]::Default) 
+    {
+        #If DHCP is not enabled, settcpipnetbios CIM Method won't take 0 so overwrite registry entry instead.
+        $RegParam = @{
+            Path = "HKLM:\SYSTEM\CurrentControlSet\Services\NetBT\Parameters\Interfaces\Tcpip_$($NetAdapterConfig.SettingID)"
+            Name = 'NetbiosOptions'
+            Value = 0
+        }
+        $null = Set-ItemProperty @RegParam
+    }
+    else
+    {
+        $null = $NetAdapterConfig | 
+            Invoke-CimMethod -MethodName settcpipnetbios -Arguments @{
+                TcpipNetbiosOptions = [uint32][NETBIOSSetting]::$Setting.value__
+            }
+    }
+}
+
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $InterfaceAlias,
+
+        [parameter(Mandatory = $true)]
+        [ValidateSet("Default","Enable","Disable")]
+        [System.String]
+        $Setting
+    )
+    $NIC = Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "NetConnectionID=`"$InterfaceAlias`""
+    if ($Null -ne $NIC)
+    {
+        Write-Verbose -Message "Interface $InterfaceAlias detected with Index number: $($NIC.InterfaceIndex)"
+    }
+    else
+    {
+        throw "Nic with Alias $InterfaceAlias was not found"

```

Comment:
Can you throw a custom error here? See this [Issue](https://github.com/PowerShell/DscResources/issues/81) for examples of what I mean. This allows us to implement unit tests for this specific error concurrence.